main.js

```diff
@@ -0,0 +1,111 @@
+const game = {
+    board: [[5, 4, 3, 2, 1], [], []],
+    peg1EmptiedOnce: false,  // Track whether peg1 has been emptied
+
+    // Display the current state of the board using map
+    displayBoard() {
+        console.log("--- Board State ---");
+        this.board.map(peg => {
+            console.log("--- " + peg.join(' '));
+        });
+    },
+
+    // Move a disc from one peg to another
+    moveDisc(fromPegNumber, toPegNumber) {
+        const fromPegIndex = fromPegNumber - 1; // Convert peg number to index (1 -> 0, 2 -> 1, etc.)
+        const toPegIndex = toPegNumber - 1;     // Convert peg number to index
+
+        const fromPeg = this.board[fromPegIndex];
+        const toPeg = this.board[toPegIndex];
+
+        // Check if fromPeg is empty
+        if (fromPeg.length === 0) {
+            console.log("No discs on this peg to move!");
+            return;
+        }
+
+        // Logic for peg1 (initial peg)
+        if (fromPegNumber === 1 && !this.peg1EmptiedOnce) {
```

Comment:
why peg 1 needs special treatment than the other pegs in terms of logic? 
I would consider revisit this approach